tests/test_predicate.py

```diff
@@ -0,0 +1,347 @@
+            query = f"2 {op} 2"
+            result = eval(query, eval_globals, eval_locals)  # skipcq: PYL-W0123
+            assert isinstance(check(result), Number)
+
+    @staticmethod
+    def test_propery_binary_operations(eval_globals, eval_locals, check):
+        """Check that binary operations between properties does not error."""
+        for op in BINARY_OP_STRINGS:
+            query = f"props['int'] {op} props['int']"
+            result = eval(query, eval_globals, eval_locals)  # skipcq: PYL-W0123
+            assert isinstance(check(result), Number)
+
+    @staticmethod
+    def test_r_binary_operations(eval_globals, eval_locals, check):
+        """Test right hand binary operations between numbers and properties."""
+        for op in BINARY_OP_STRINGS:
+            query = f"2 {op} props['int']"
+            result = eval(query, eval_globals, eval_locals)  # skipcq: PYL-W0123
+            assert isinstance(check(result), Number)
+
+    @staticmethod
+    def test_number_prefix_operations(eval_globals, eval_locals, check):
+        """Test prefix operations on numbers."""
+        for op in PREFIX_OP_STRINGS:
+            query = f"{op}1"
+            result = eval(query, eval_globals, eval_locals)  # skipcq: PYL-W0123
+            assert isinstance(check(result), Number)
+
+    @staticmethod
+    def test_property_prefix_operations(eval_globals, eval_locals, check):
+        """Test prefix operations on properties."""
+        for op in PREFIX_OP_STRINGS:
+            query = f"{op}props['int']"
+            result = eval(query, eval_globals, eval_locals)  # skipcq: PYL-W0123
+            assert isinstance(check(result), Number)
+
+    @staticmethod
+    def test_regex_nested_props(eval_globals, eval_locals, check):
+        """Test regex on nested properties."""
+        query = "props['nesting']['fib'][4]"
+        result = eval(query, eval_globals, eval_locals)  # skipcq: PYL-W0123
+        assert check(result) == 5
+
+    @staticmethod
+    def test_regex_str_props(eval_globals, eval_locals, check):
+        """Test regex on string properties."""
+        query = "regexp('Hello', props['string'])"
+        result = eval(query, eval_globals, eval_locals)  # skipcq: PYL-W0123
+        assert check(result) == "Hello"
+
+    @staticmethod
+    def test_regex_str_str(eval_globals, eval_locals, check):
+        """Test regex on string and string."""
+        query = "regexp('Hello', 'Hello world!')"
+        result = eval(query, eval_globals, eval_locals)  # skipcq: PYL-W0123
+        assert check(result) == "Hello"
+
+    @staticmethod
+    def test_regex_props_str(eval_globals, eval_locals, check):
+        """Test regex on property and string."""
+        query = "regexp(props['string'], 'Hello world!')"
+        result = eval(query, eval_globals, eval_locals)  # skipcq: PYL-W0123
+        assert check(result) == "Hello world!"
+
+    @staticmethod
+    def test_regex_ignorecase(eval_globals, eval_locals, check):
```

Comment:
test_regex_ignore_case

Reply:
This is referring to the enum which is one word